tools/parse/src/main/scala/org/allenai/nlpstack/parse/poly/ml/GoogleNGram.scala

```diff
@@ -0,0 +1,223 @@
+    */
+  private def parseLine(line: String): (String, Seq[SyntacticInfo], Long) = {
+    line.split("\t").map(x => x.trim) match {
+      case Array(ngram: String, syntacticNgramStr: String, freqStr: String, _*) =>
+        (ngram.toLowerCase, createSyntacticNgram(syntacticNgramStr), freqStr.toLong)
+    }
+  }
+}
+
+/** Encapsulates unigram info pertaining to a word. Instead of a seq of SyntacticInfo objects in the
+  * general purpose NgramInfo class, here we have a single SyntacticInfo representing the info for
+  * a single gram.
+  */
+case class UnigramInfo(syntacticUnigram: SyntacticInfo, frequency: Long)
+
+/** Object encapsulating some functionality specific to unigrams. Used wherever features need to be
+  * constructed based on unigrams (Google Ngram Nodes).
+  */
+object GoogleUnigram {
+
+  /** Looks up specified ngramMap for the given token and returns a map of the frequency for each
+    * dependency label for the given token word and POS, normalized over the total frequency for all
+    * possible dependency labels.
+    * @param token the token to look up
+    * @param ngramMap the table mapping a word to the sequence of NgramInfos, as obtained from the
+    * GoogleNGram class object
+    * @param frequencyCutoff the frequency cutoff that was used to construct the map. This is used
+    * here to shift the scale of the frequencies to start from the cutoff point instead of 1.
+    */
+  def getDepLabelNormalizedDistribution(
+    token: PostaggedToken, ngramMap: Map[String, Seq[NgramInfo]], frequencyCutoff: Int
+  ): Map[String, Double] = {
+    val tokenNodeInfos = (for {
+      tokNgrams <- ngramMap.get(token.string.toLowerCase)
+    } yield {
+      getTokenUnigramInfo(
+        Option(token.postag), tokNgrams, frequencyCutoff
+      )
+    }).getOrElse(Seq.empty[UnigramInfo])
+
+    // Get the total frequency for all nodes aggregated above for the current token to
+    // normalize them.
+    val totalFrequency = tokenNodeInfos.foldLeft(0L)((a, b) => a + b.frequency)
+
+    // Iterate over all the nodes, and normalize frequencies by the total frequency.
+    (for {
+      tokenNodeInfo <- tokenNodeInfos
+    } yield {
+      val normalizedFrequency = tokenNodeInfo.frequency.toDouble / totalFrequency
+      (tokenNodeInfo.syntacticUnigram.depLabel -> normalizedFrequency)
+    }).toMap
+  }
+
+  /** Looks up specified ngramMap for the given word and returns a map of the frequency for each
+    * POS tag for the given word, normalized over the total frequency for all possible POS tags.
+    * @param word the word to look up
+    * @param ngramMap the table mapping a word to the sequence of NgramInfos, as obtained from the
+    * GoogleNGram class object
+    * @param frequencyCutoff the frequency cutoff that was used to construct the map. This is used
+    * here to shift the scale of the frequencies to start from the cutoff point instead of 1.
+    */
+  def getPosTagNormalizedDistribution(
```

Comment:
Actually they are not the same- they pass different first augments to `getTokenUnigramInfo`. The later 'for` comprehensions are also doing different things.